### PR TITLE
dagent plugin for moving to tag folder

### PR DESCRIPTION
**Pull request by m42e**
If you have a mail address with a tag like in gmail and a folder
with the tags name the mail will be moved to that folder. It
can also create the folder if you like. Else a tag will be added to the
subject.

### CODE REVIEW

**Comment by fbartels** [comment]
Hello Matthias,

I get the following error in dagent.log when enabling this plugin:
```
Thu May 14 20:27:02 2015: [14562] !-- error: list index out of range 
Thu May 14 20:27:02 2015: [14562] !- Plugin 'moveToKeyword.PreDelivery' call ignored please check the plugin
```

Can you also add a description to the readme.md file? Optimally this description should include examples what tags are (link would be enough) and how the plugin can be configured (if a folder should be created or the tag should be written to the subject), so that anyone reading the readme can get an impression what you plugin can do for them.

Regards

**Comment by m42e** [comment]
Ok, hmmm, somehow I failed in copy the working version. DOH!
Give it another try, and take a look at the readme.

**Comment by fbartels** [comment]
Seems to work for me. @jelly @bkram what do you think?

**Comment by m42e** [comment]
Give me another day to apply a minor fix for domains containing a dash.

File: plugins/moveToTag.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from MAPI.Time import *
+from MAPI.Struct import *
+
+import pprint
+
+from plugintemplates import *
+
+class moveToTag(IMapiDAgentPlugin):
+
+    createFolder = 0
```
Comment (jelly) [inline plugins/moveToTag.py]: I would use False or True here instead of 0 or 1. Booleans make the code more readable

File: plugins/moveToTag.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            return MP_CONTINUE
+        elements = re.split('-|\+|@', mailto)
+        # here we should have three elements:
+        # - the receiver
+        # - the tag <-- thats what we interested in
+        # - the domain
+        if len(elements) < 3:
+            return MP_CONTINUE
+        tag = elements[1]
+        found = 0
```
Comment (jelly) [inline plugins/moveToTag.py]: Same comment as above :)

File: plugins/moveToTag.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            store.NotifyNewMail( NEWMAIL_NOTIFICATION(msgid, folderid, 0, None, 0) )
+            # stop here, we already delivered the message successfully
+            return MP_STOP_SUCCESS
+
+        # in case we do not have a folder and do not want to create one, 
+        # we add the tag to the subject
+        subject = message.GetProps([PR_SUBJECT],0)[0].Value
+        subject = "%s [%s]" % (subject, tag)
+        message.SetProps([SPropValue(PR_SUBJECT, subject)])
+        return MP_CONTINUE
```
Comment (jelly) [inline plugins/moveToTag.py]: I don't quiet follow the flow here. Wouldn't it make sense to first find the folder with the tag, if that does not exists create one if the user configures this and then copy the message.

I think you should be able to use a try catch for finding the folder and in the catch you can create the folder if this is required. (If I find time to configure my postfix and setup this plugin I'll see if I can write a patch

**Pull request closed without merging**